program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,76 @@
 # Create your views here.
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from core.models import ProgramIntent
+from api.serializers import ProgramIntentSerializer
+from edx_api_doc_tools import path_parameter, schema
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework.views import APIView
+
+# TODO not sure what this name should be
+class ProgramRetrieveView(viewsets.ModelViewSet):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent CERTAIN intent
+    Path: /api/[version]/programs/{user}
+    Returns:
+     * 200: OK
+     * 404: Program does not exist.
+    """
+    serializer_class = ProgramIntentSerializer
+    permission_required = [perms.API_READ_METADATA]
+    event_method_map = {'GET': 'registrar.{api_version}.get_program_detail'}
+    event_parameter_map = {'program_key': 'program_key'}
+    # TODO unsure what this means ^
+
+    def get_object(self):
+        """
+        Get the Program to be serialized and returned.
+        Overrides RetrieveAPIView.get_object.
+        """
+        return self.program
+    # unsure if i need this ^
+
+    def get_serializer_context(self):
+        """
+        Get the extra data needed to serialize the Program.
+        Overrides ListAPIView.get_serializer_context.
+        """
+        return {
+            **super().get_serializer_context(),
+            "user_api_permissions_by_program": {
+                self.program: get_api_permissions_by_program(
+                    user=self.request.user, program=self.program
+                ),
+            },
+        }
```

Comment:
not sure if these are necessary, but added since I saw it in exams codebase

Reply:
nvm i realize that this was yoinked from the registrar repo. I believe these can be removed since that was for a different context?